app/code/Magento/ImportExport/Model/Import/Source/FileParser/UnsupportedPathException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+class UnsupportedPathException extends \RuntimeException
+{
+    private $path;
+
+    public function __construct($path = '', $message = '')
+    {
+        parent::__construct($message ?: sprintf('Path "%s" is not supported', $path));
+
+        $this->path = $path;
+    }
+
+    public function getPath()
```

Comment:
I think you should consider different presentations layers, not only web. 
Mapping to LocalizedException doesn't mean just wrapping exception `getPath()` method. Mapper must prepare exception for specific presentation layer depending on its context. If path revealing is a bad idea for web UI, but there are valid use cases for shell application.
For instance, if I start importing multiple files from a shell script, I want to know complete path to a failed file to import. 

It is just a nice feature to have. If you recommend not having it, I am fine with removing it.